src/renderer/services/BeaconAPIClient.ts

```diff
@@ -0,0 +1,226 @@
+            return result;
+        } catch (error) {
+            return error.response.status;
+        }
+    }
+
+    async publishSignedBlock(beacon_block: BeaconBlock) {
+        // TODO handle error catch
+        try {
+            const result = await this.genericPost<any>(PUBLISH_SIGNED_BLOCK(beacon_block));
+            return result.status;
+        } catch (error) {
+            return error.response.status;
+        }
+    }
+
+    async produceAttestation(validator_pubkey: BLSPubkey, poc_bit: uint8, slot: Slot, shard: Shard) {
+        // TODO handle error catch
+        try {
+            const result = await this.genericFetch<IndexedAttestation>(
+                PRODUCE_ATTESTATION(validator_pubkey, poc_bit, slot, shard)
+            );
+            return result;
+        } catch (error) {
+            return error.response.status;
+        }
+    }
+
+    async publishSignedAttestation(attestation: IndexedAttestation) {
+        // TODO handle error catch
+        try {
+            const result = await this.genericPost<any>(PUBLISH_SIGNED_ATTESTATION(attestation));
+            return result.status;
+        } catch (error) {
+            return error.response.status;
+        }
+    }
+
+    /**
+     * Function that encapsulate executing GET method and returns Promise with generic type
+     * @param url endpoint for fetch
+     */
+    async genericFetch<T>(url: string): Promise<T> {
+        let result: T;
+
+        const promise: Promise<T> = new Promise((resolve, reject) => {
+            this.fetchFromApi(url, (err: any, response: any) => {
+                if (!err) {
+                    resolve(response.data);
+                } else {
+                    reject(err);
+                }
+            });
+        });
+
+        // wait until the promise returns us a value
+        result = await promise;
+        return result;
+    }
+
+    /**
+     * Function that encapsulate executing POST method and returns Promise with generic type
+     * @param url endpoint for fetch
+     * @param data body for POST method
+     */
+    async genericPost<T>(url: string, ...data: any[]): Promise<T> {
+        let result: T;
+
+        const promise: Promise<T> = new Promise((resolve, reject) => {
+            this.postToApi(url, data, (err: any, response: any) => {
+                if (!err) {
+                    resolve(response);
+                } else {
+                    reject(err);
+                }
+            });
+        });
+
+        // wait until the promise returns us a value
+        result = await promise;
+        return result;
+    }
+
+    /**
+     * Helper function for POST method
+     * @param url endpoint url
+     * @param data request body
+     * @param callback callback function that will return either response or error
+     */
+    postToApi = (url: string, data: any = {}, callback: Function) => {
```

Comment:
It might be better to extract those generic http api methods to separate class like here:
https://github.com/ChainSafe/lodestar/blob/master/packages/lodestar/src/util/httpClient.ts